octobot/strategy_optimizer/optimizer_settings.py

```diff
@@ -21,72 +21,142 @@
+import octobot_backtesting.constants as backtesting_constants
 
 
 class OptimizerSettings:
     def __init__(self, settings_dict=None):
         if settings_dict is None:
             settings_dict = {}
         # generic
-        self.optimizer_config = settings_dict.get(enums.OptimizerConfig.OPTIMIZER_CONFIG.value, None)
-        self.randomly_chose_runs = settings_dict.get(enums.OptimizerConfig.RANDOMLY_CHOSE_RUNS.value,
-                                                     constants.OPTIMIZER_DEFAULT_RANDOMLY_CHOSE_RUNS)
-        self.data_files = settings_dict.get(enums.OptimizerConfig.DATA_FILES.value)
-        self.start_timestamp = settings_dict.get(enums.OptimizerConfig.START_TIMESTAMP.value, None)
-        self.end_timestamp = settings_dict.get(enums.OptimizerConfig.END_TIMESTAMP.value, None)
-        self.required_idle_cores = int(settings_dict.get(enums.OptimizerConfig.IDLE_CORES.value,
-                                                         constants.OPTIMIZER_DEFAULT_REQUIRED_IDLE_CORES))
-        self.notify_when_complete = settings_dict.get(enums.OptimizerConfig.NOTIFY_WHEN_COMPLETE.value,
-                                                      constants.OPTIMIZER_DEFAULT_NOTIFY_WHEN_COMPLETE)
-        self.optimizer_mode = settings_dict.get(enums.OptimizerConfig.MODE.value,
-                                                enums.OptimizerModes.NORMAL.value)
+        self.exchange_ids = settings_dict.get(
+            enums.OptimizerConfig.EXCHANGE_IDS.value,
+        )
+        if not self.exchange_ids:
+            self.exchange_ids = [
+                settings_dict.get(enums.OptimizerConfig.EXCHANGE_ID.value)
+            ]
+
+        self.exchange_type = settings_dict.get(
+            enums.OptimizerConfig.EXCHANGE_TYPE.value
+        )
+        self.optimizer_config = (
+            settings_dict.get(enums.OptimizerConfig.OPTIMIZER_CONFIG.value) or None
+        )
+        self.randomly_chose_runs = (
+            settings_dict.get(enums.OptimizerConfig.RANDOMLY_CHOSE_RUNS.value)
+            or constants.OPTIMIZER_DEFAULT_RANDOMLY_CHOSE_RUNS
+        )
+
+        self.data_files = settings_dict.get(
+            enums.OptimizerConfig.DATA_FILES.value,
+            [backtesting_constants.CONFIG_CURRENT_BOT_DATA],
+        )
+        if not isinstance(self.data_files, list):
+            self.data_files = [self.data_files]
+        if backtesting_constants.CONFIG_CURRENT_BOT_DATA in self.data_files:
+            self.data_files = [backtesting_constants.CONFIG_CURRENT_BOT_DATA]
+        self.start_timestamp = settings_dict.get(
+            enums.OptimizerConfig.START_TIMESTAMP.value, None
+        )
+        self.end_timestamp = settings_dict.get(
+            enums.OptimizerConfig.END_TIMESTAMP.value, None
+        )
+        self.required_idle_cores = int(
+            settings_dict.get(enums.OptimizerConfig.IDLE_CORES.value)
+            or constants.OPTIMIZER_DEFAULT_REQUIRED_IDLE_CORES
+        )
```

Comment:
I think using `or` when assigning default values from `settings_dict` can cause incorrect defaults when valid falsy values (e.g., `0`, `False`) are provided. It might be better to change this to:
```python
self.required_idle_cores = int(
    settings_dict.get(
        enums.OptimizerConfig.IDLE_CORES.value,
        constants.OPTIMIZER_DEFAULT_REQUIRED_IDLE_CORES
    )
)
```

Reply:
Hi @ruidazeng 
I see, I have removed those changes.

Reply:
@Herklos is this something merge-able? trying to backlog all the pull requests for OctoBot :))

Reply:
@ruidazeng Thank you but we can see the open pull requests on the OctoBot repos, can you please not do this?